src/lib/wallet/GoodWalletClass.js

```diff
@@ -438,21 +438,47 @@ export class GoodWallet {
     )
   }
 
+  async processEvents(eventsPromise, startBlock) {
+    const events = await eventsPromise
+    const chunks = chunk(events, 10)
+    const limit = pRateLimit({ concurrency: 2, interval: 1000, rate: 1 })
+    await Promise.all(chunks.map(c => limit(() => this._notifyEvents(c, startBlock))))
+  }
+
   async syncTxFromExplorer(startBlock) {
-    let results = []
-    const { tokenContract, account, networkId } = this
+    const { account, networkId, tokenContract, oneTimePaymentsContract } = this
     const { _address: tokenAddress } = tokenContract || {}
+    const { _address: otpAddress } = oneTimePaymentsContract || {}
 
-    if (tokenAddress) {
-      results = await API.getTokenTxs(tokenAddress, account, networkId, startBlock).then(results =>
-        results.map(result => ({ ...result, transactionHash: result.hash })),
-      )
+    // this is both send/receive events
+    let results = []
+    let otpResults = []
+    const tokenPromise = tokenAddress
+      ? API.getTokenTxs(tokenAddress, account, networkId, startBlock).then(results =>
+          results.map(result => ({ ...result, transactionHash: result.hash })),
+        )
+      : Promise.resolve()
+
+    const withdrawHash = '0x39ca68a9f5d8038e871ef25a6622a56579cda4a6eedf63813574d23652e94048'
+    const cancelHash = '0xb1f6a8f6b8fb527cfeec0df589160bc2a92ff715097117cb250e823c7106bc2f'
+
+    const otpPromise = otpAddress
+      ? Promise.all([
+          API.getOTPLEvents(account, networkId, otpAddress, startBlock, withdrawHash).then(results =>
+            results.map(result => ({ ...result, transactionHash: result.hash })),
+          ),
+          API.getOTPLEvents(account, networkId, otpAddress, startBlock, cancelHash).then(results =>
+            results.map(result => ({ ...result, transactionHash: result.hash })),
+          ),
+        ])
+      : Promise.resolve()
 
-      const limit = pRateLimit({ concurrency: 2, interval: 1000, rate: 1 })
-      const chunks = chunk(results, 10)
+    ;[results, otpResults] = await Promise.all([tokenPromise, otpPromise])
 
-      await Promise.all(chunks.map(c => limit(() => this._notifyEvents(c, startBlock))))
-    }
+    await Promise.all([
```

Comment:
why not simply merge results+otpresults as you run the the exact same method on them.

Reply:
Depends. if we keep it like it is now? yes it can be merged
but as mentioned on monday call, if we wish to handle the case that if one of the api-calls is failing only the failing one should be retrieved through bc call. then this still makes sense